src/Symfony/Component/Routing/Matcher/Dumper/PhpMatcherDumper.php

```diff
@@ -244,9 +244,14 @@ private function compileRoute(Route $route, $name, $supportsRedirections, $paren
                 $regex = substr($regex, 0, $pos).'/?$'.substr($regex, $pos + 2);
                 $hasTrailingSlash = true;
             }
-            $conditions[] = sprintf('preg_match(%s, $pathinfo, $matches)', var_export($regex, true));
 
-            $matches = true;
+            if (!preg_match('#^'.preg_quote('#^/(?P<', '#').'([A-Za-z0-9_]+)'.preg_quote('>.*)?$#s', '#').'$#', $regex)) {
```

Comment:
This is more like a proof-of-concept, as I have mentioned - I just looked at the regex which the Router creates for the catch-all route and matched it, and it works. An example of the regex with the placeholder "url" would be `#^/(?P<url>.*)?$#s`, so it looks for `#^/(?P<` then a placeholder variable name and then `>.*)?$#s`.